src/backend/distributed/utils/citus_stat_tenants.c

```diff
@@ -605,8 +608,13 @@ ReduceScoreIfNecessary(TenantStats *tenantStats, TimestampTz queryTime)
 	 */
 	if (periodCountAfterLastScoreReduction > 0)
 	{
-		tenantStats->score >>= periodCountAfterLastScoreReduction;
 		tenantStats->lastScoreReduction = queryTime;
+
+		/*addtional check to avoid undefined behavior */
+		tenantStats->score = (periodCountAfterLastScoreReduction <
+							  TENANTS_STATS_FIELD_BIT_LENGTH(tenantStats, score))
+			? tenantStats->score >> periodCountAfterLastScoreReduction
+			: 0;
```

Comment:
```suggestion
		/* addtional check to avoid undefined behavior */
		tenantStats->score = (periodCountAfterLastScoreReduction <
							  TENANTS_STATS_SCORE_FIELD_BIT_LENGTH)
			? tenantStats->score >> periodCountAfterLastScoreReduction
			: 0;
```